### ISSUE TITLE
Positive results

### ISSUE THREAD

**Issue by neosr-project** [mandalinadagi/WGSR#5]
Hi, wanted to share some positive results using your research. I took the liberty to [simplify](https://github.com/muslll/neosr/commit/930ca73dbfb0dc889256267a10153ab189176b1a) your loss implementation into a single file, in [neosr](https://github.com/muslll/neosr/).

Left is *without*, Right is *with* Wavelet-Guided loss. All tests performed using manual seed and torch deterministic algorithms (making it bitwise reproducible). First few iters has instabilities, but it soon recovers and stabilizes gan. Convergence video:
https://github.com/mandalinadagi/WGSR/assets/132400428/cf0da31e-70c0-4e76-8d76-3ad6d3707da7

Side-by-side:
![wg_comparison](https://github.com/mandalinadagi/WGSR/assets/132400428/107417ed-78d9-4581-b6f0-ad542fde1779)

**Comment by mandalinadagi** [comment]
Thank you so much for your work, are those results generated from bicubic downsampled images or from real-world samples?

**Comment by neosr-project** [comment]
Those tests are on bicubic degraded, yes. We have also [tested it in real-world SR](https://github.com/LabShuHangGU/Adaptive-Token-Dictionary/issues/1#issuecomment-2015583503) with positive results :+1: 

@Phhofm finetuned a ATD model using it

**Comment by mandalinadagi** [comment]
Great, I will be adding those results and give reference to your repo for enabling the utilization of our wavelet domain loss with different methods. Thanks again, cheers!

**Comment by neosr-project** [comment]
Please feel free to use however you wish, thanks. Here are the original images: [wg_test.zip](https://github.com/mandalinadagi/WGSR/files/14760383/wg_test.zip). And models (SPAN network, inference can be done using test.py with [this configuration](https://github.com/muslll/neosr/blob/master/options/test_span.yml)): [models.zip](https://github.com/mandalinadagi/WGSR/files/14760410/models.zip)

**Comment by Phhofm** [comment]
I had also seen positive results of handling output artifacts using this research while training a dat2 model. The following are examples of where I first trained for 90k iters, then enabled the wavelet-guided loss in neosr (only change i made), then continued training until 190k iters.

[Slowpics slider comparison, 3 examples](https://slow.pics/s/wo4lgWek)

![wavelet_example1](https://github.com/mandalinadagi/WGSR/assets/14755670/904aa931-afa8-4789-a394-28fc34892b4a)
![wavelet_example2](https://github.com/mandalinadagi/WGSR/assets/14755670/6ed527de-6db3-46bd-8311-4ea04f7354e3)
![wavelet_example3](https://github.com/mandalinadagi/WGSR/assets/14755670/398ef117-b863-4751-b4fb-adde45cf43a9)

**Comment by mandalinadagi** [comment]
great news, thank you!

**Comment by caijie0620** [comment]
> Hi, wanted to share some positive results using your research. I took the liberty to [simplify](https://github.com/muslll/neosr/commit/930ca73dbfb0dc889256267a10153ab189176b1a) your loss implementation into a single file, in [neosr](https://github.com/muslll/neosr/).
> 
> Left is _without_, Right is _with_ Wavelet-Guided loss. All tests performed using manual seed and torch deterministic algorithms (making it bitwise reproducible). First few iters has instabilities, but it soon recovers and stabilizes gan. Convergence video: https://github.com/mandalinadagi/WGSR/assets/132400428/cf0da31e-70c0-4e76-8d76-3ad6d3707da7
> 
> Side-by-side: ![wg_comparison](https://private-user-images.githubusercontent.com/132400428/316736985-107417ed-78d9-4581-b6f0-ad542fde1779.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxODM0NTIsIm5iZiI6MTcxMjE4MzE1MiwicGF0aCI6Ii8xMzI0MDA0MjgvMzE2NzM2OTg1LTEwNzQxN2VkLTc4ZDktNDU4MS1iNmYwLWFkNTQyZmRlMTc3OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwM1QyMjI1NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00ZDE2Y2Y0NGU5NjkzODI2NjJiNDgyYjhmMzJjMTRkOTg3OTIzZTNhMzY2NmQ1NzQ5MDEwNGUyMDE2NDA4NmU0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.J3fZna33n_KyWxqdTYSVoNC9FlGyX0bEFJhEfSYAZ0o)

Why do I think the left looks more visually appealing? The left one has more artifacts.

**Comment by caijie0620** [comment]
> Hi, wanted to share some positive results using your research. I took the liberty to [simplify](https://github.com/muslll/neosr/commit/930ca73dbfb0dc889256267a10153ab189176b1a) your loss implementation into a single file, in [neosr](https://github.com/muslll/neosr/).
> 
> Left is _without_, Right is _with_ Wavelet-Guided loss. All tests performed using manual seed and torch deterministic algorithms (making it bitwise reproducible). First few iters has instabilities, but it soon recovers and stabilizes gan. Convergence video: https://github.com/mandalinadagi/WGSR/assets/132400428/cf0da31e-70c0-4e76-8d76-3ad6d3707da7
> 
> Side-by-side: ![wg_comparison](https://private-user-images.githubusercontent.com/132400428/316736985-107417ed-78d9-4581-b6f0-ad542fde1779.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTIxODM3NDgsIm5iZiI6MTcxMjE4MzQ0OCwicGF0aCI6Ii8xMzI0MDA0MjgvMzE2NzM2OTg1LTEwNzQxN2VkLTc4ZDktNDU4MS1iNmYwLWFkNTQyZmRlMTc3OS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNDAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDQwM1QyMjMwNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMjVjYmZhZDVmOGIxYTFmOTVkMDU4MWE4MmVlMTVlN2QwNTdjOWMyNDg0M2I2MDAwNWFlNTdjYzIzMGE4NTY4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NqhVmazn-Rk2lnnpN_6_QvhueDsOWlu1xugcSMhqdjE)

Why do I think the left looks more visually appealing? The right one has more artifacts.

**Comment by neosr-project** [comment]
> Why do I think the left looks more visually appealing? The right one has more artifacts.

Baseline is sharper, doesn't mean it has more 'details'. Wavelet decreased the artifacts created by GAN training, which is the purpose of this research. It's visible on the t-shirt on the first sample.
